File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        number = str(x)
        start, end = 0, len(number) - 1

        while start < end:
            if number[start] != number[end]:
                return False
            start += 1
            end -= 1

        return True

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        String number = Integer.toString(x);
        int start = 0, end = number.length() - 1;

        while (start < end) {
            if (number.charAt(start) != number.charAt(end)) {
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
}